### DISCUSSION TITLE
考虑支持对fastly ipv6的优化吗

### DISCUSSION THREAD

**Discussion by zhengqwe** [pymumu/smartdns#1835] [General]
方法参考[这个](https://github.com/IrineSistiana/mosdns/discussions/511)，简单说就是两边不变换中间，应该比较好实现

**Comment by PikuZheng** [comment]
我这边fastly如果不过墙 国内ip会被ban。我自己是没法用fastly和akamai的优选的。

**Reply by zhengqwe** [reply]
> 我这边fastly如果不过墙 国内ip会被ban。我自己是没法用fastly和akamai的优选的。

我这边v4有你这样的情况，但v6很稳

**Reply by PikuZheng** [reply]
如果是一个ip段换成一个固
定的ip的话可以用ip-alias。
但是上面那个discussion看起来是要做4to6，这不太可行，因为v4 v6本质上是两个不同的查询，不可能查询v4时给出个v6结果

**Reply by zhengqwe** [reply]
> 如果是一个ip段换成一个固 定的ip的话可以用ip-alias。 但是上面那个discussion看起来是要做4to6，这不太可行，因为v4 v6本质上是两个不同的查询，不可能查询v4时给出个v6结果

不是4to6，是只对v6地址选区，v4不处理，例子：`2a04:4e42:400::313`换成`2a04:4e42:36::313`，`2a04:4e42:200::347`换成`2a04:4e42:36::347`
另外，ip-alias只适合只需固定前缀，后缀随便的情况，而fastly前后缀都要固定，只有中间能换，如上面的400，200和36

**Reply by zhengqwe** [reply]
想到一个比较好的实现方式：增加一个选项，例如`prefix-replace`，它有两个参数，第一个是匹配条件，第二个是替换目标，上面的情况就可以写成`prefix-replace 2a04:4e42::/32 2a04:4e42:36::/48`，并且这个选项还能代替`ip-alias`，如文档中的`ip-alias 1.2.3.4/16 4.5.6.7`可以写成`prefix-replace 1.2.3.4/16 4.5.6.7/32`
